### ISSUE TITLE
XML: add ability to provide a specific XMLOutputFactory

### ISSUE THREAD

**Issue by jphelp32** [akka/alpakka#1936]
### Short description

Currently, the Xml Writer can only take the default XMLOutputFactory as configured for the JVM instance. It would be helpful if one could provide an alternate XMLOutputFactory to be used for the Xml Writer, rather than always having to take the default.

### Details

Should be fairly simple to add XMLOutputFactory as an arg to the `StreamingXmlWriter` constructor, and then add another method to `XmlWriting` as follows:
```
def writer(charset: Charset, outputFactory: XMLOutputFactory): Flow[ParseEvent, ByteString, NotUsed] =
    Flow.fromGraph(new impl.StreamingXmlWriter(charset, outputFactory))
```

**Comment by ennru** [comment]
Hi @jphelp32 	
Thank you for this useful suggestion. Would you be in a position to suggest the necessary changes in a Pull Request?

**Comment by jphelp32** [comment]
@ennru I'm inquiring w/my employer re: the CLA..  Not sure how difficult it will be to get that part of it done.
In the meantime I've forked alpakka and pulled it down, but having some trouble importing the project into IDEA and being able to build/test/etc. I'm not too familiar with sbt projects unfortunately.

Side note, there's a similar issue here on the parsing side. I know that it relies upon the aalto parser, but currently there is no way to set any specific configuration on that factory. And by default, the factory configuration supports DTD and external entities, which is not secure. Should I log a separate issue for this or could that be added into this one?

**Comment by ennru** [comment]
If you have sbt installed it might help to run `sbt Test/update` outside of IntelliJ before importing. You could comment out the other modules in `build.sbt` before importing.

Please create a separate issue for the XML parser.

**Comment by ennru** [comment]
Fixed with #1945
